src/com/hotmail/kalebmarc/textfighter/main/Weapon.java

```diff
@@ -56,6 +57,32 @@ public Weapon(String name, int ammoUsed, int ammoIncludedWithPurchase, boolean b
 
     }
 
+    // used to set the criticalChance to non default value of .01
+    public Weapon(String name, int ammoUsed, int ammoIncludedWithPurchase, boolean buyable, int price, //For guns
+                  int ammoPrice, int level, double chanceOfMissing, boolean firstInit, boolean changeDif, double criticalChance) {
+
+        this.name = name;
+        this.ammoUsed = ammoUsed;
+        this.ammoIncludedWithPurchase = ammoIncludedWithPurchase;
+        this.buyable = buyable;
+        this.price = price;
+        this.ammoPrice = ammoPrice;
+        this.level = level;
+        this.chanceOfMissing = chanceOfMissing;
+        this.melee = false;
+        this.criticalChance = criticalChance;
+
+        if (!changeDif) {

```

Comment:
```suggestion
        if (!changeDif) arrayWeapon.add(this);
```

Reply:
these two if statements here can be done in one line each :) This is just a suggestion, its completely up to you if you want to do this 👍